### PR TITLE
First  Goodie hack

### PR DESCRIPTION
**Pull request by abhijainn27**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
It prints my github username and tell that  am awesome and completed the goodie hack.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It helps to find me among all other guthub users

**What is the data source for your Instant Answer? (Provide a link if possible)**
https://duck.co/duckduckhack/goodie_quickstart

**Why did you choose this data source?**
Because it is my first goodie hack

**Are there any other alternative (better) data sources?**

**What are some example queries that trigger this Instant Answer?**
duckduckhack abhijainn27

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
github users

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

**Which existing Instant Answers will this one supercede/overlap with?**
None

**Are you having any problems? Do you need our help with anything?**
No

**Where did you hear about DuckDuckHack? (For first time contributors)**
FSMK

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
ddg

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [x] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 10

    - Windows 7
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] IE 8
        - [] IE 9
        - [] IE 10

    - Windows XP
        - [] IE 7
        - [] IE 8

    - Mac OSX
        - [] Google Chrome
        - [] Firefox
        - [] Opera
        - [] Safari

    - iOS (iPhone)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - iOS (iPad)
        - [] Safari Mobile
        - [] Google Chrome
        - [] Opera

    - Android
        - [] Firefox
        - [] Native Browser
        - [] Google Chrome
        - [] Opera

    - Linux (Ubuntu)
        - [x] Firefox

```

### CODE REVIEW

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+use DDG::Goodie;
+
+zci answer_type => "is_awesome_abhijainn27";
+zci is_cached   => 1;
+
+# Metadata.  See https://duck.co/duckduckhack/metadata for help in filling out this section.
+name "IsAwesome abhijainn27";
+description "Succinct explanation of what this instant answer does";
+primary_example_queries "first example query", "second example query";
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: @abhijainn27 Could you change this to be the actual query the goodie will trigger with? In this case that would be "duckduckhack abhijainn27". Thanks!

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+secondary_example_queries "optional -- demonstrate any additional triggers";
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: @abhijainn27 You can remove this line - in this case we don't have any additional triggers :)

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# to instant answer development
+
+use DDG::Goodie;
+
+zci answer_type => "is_awesome_abhijainn27";
+zci is_cached   => 1;
+
+# Metadata.  See https://duck.co/duckduckhack/metadata for help in filling out this section.
+name "IsAwesome abhijainn27";
+description "Succinct explanation of what this instant answer does";
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: @abhijainn27 You can have a custom description, too! Like "tell the world abhijainn27 is awesome" :)

**Comment by MariagraziaAlastra** [comment]
@abhijainn27 Thanks for this! I've added a few comments inline, just minor things. Nice work!

**Comment by abhijainn27** [comment]
Hey @MariagraziaAlastra I have made the above changes.. Should i send a new pull request ?

**Comment by MrChrisW** [comment]
@abhijainn27 Provided you push to your master branch, changes you make will be added to this PR.

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+primary_example_queries "duckduckhack abhijainn27";secondary_example_queries "optional -- demonstrate any additional triggers";
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: You can remove `secondary_example_queries "optional -- demonstrate any additional triggers";`

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#category
+# category "";
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#topics
+# topics "";
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: Please remove all comments, these are for tutorial purposes only.

**Comment by abhijainn27** [comment]
Hey @MrChrisW I removed the secondary query and comment lines. Hope this works fine now. :)

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+triggers start => "duckduckhack abhijainn27";
+
+
+handle remainder => sub {
+    return if $_;
+	return "abhijainn27 is a DDG user and has successfully completed the DuckDuckHack Goodie tutorial!";
+
+	return unless $_; # Guard against "no answer"
+
+	return $_;
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: These two `return` statements are not needed. They'll never be reached as there's a return on line 24

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+zci answer_type => "is_awesome_abhijainn27";
+zci is_cached   => 1;
+
+
+name "IsAwesome abhijainn27";
+description "Succinct explanation of what this instant answer does";
+primary_example_queries "duckduckhack abhijainn27";
+
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/abhijainn27.pm";
+attribution github => ["GitHubAccount", "Friendly Name"],
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: This should be `attribution github => ["https://github.com/abhijainn27", "abhijainn27"];`

**Comment by abhijainn27** [comment]
Hey @MrChrisW  I made the above changes.

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package DDG::Goodie::IsAwesome::abhijainn27;
+
```
Comment (javathunderman) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: Please add an abstract (something like https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/IsAwesome/javathunderman.pm#L2)

**Comment by javathunderman** [comment]
Hey @abhijainn27, I left a line note for you. Please fix this and it should be ready to merge. 

cc @MrChrisW

**Comment by abhijainn27** [comment]
Hey @javathunderman I added the abstract.

**Comment by javathunderman** [comment]
@MrChrisW This LGTM...I suggest you give a final review.

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+use DDG::Goodie;
+
+zci answer_type => "is_awesome_abhijainn27";
+zci is_cached   => 1;
+
+
+name "IsAwesome abhijainn27";
+description "My First Goodie Hack !! ";
+primary_example_queries "duckduckhack abhijainn27";
+
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: @abhijainn27 One last thing, please add 
````perl
category "special";
topics "special_interest", "geek";
````

Like [this](https://github.com/abhijainn27/zeroclickinfo-goodies/blob/953ab413eb8bb0f9dc6f51f2c5382fb5558f7b58/lib/DDG/Goodie/IsAwesome/NicolasGeraud.pm#L12-L13)

**Comment by abhijainn27** [comment]
Done :)

File: lib/DDG/Goodie/IsAwesome/abhijainn27.pm [hunk 10]
```diff
@@ -0,0 +1,29 @@
+package DDG::Goodie::IsAwesome::abhijainn27;
+# ABSTRACT: abhijainn27 first Goodie Hack !!
+# CATEGORY "DuckDuckHack Goodie";
+# TOPICS "Web Development", "geek";
```
Comment (MrChrisW) [inline lib/DDG/Goodie/IsAwesome/abhijainn27.pm]: @abhijainn27 Sorry for the delay. 

You'll need to add these **on line 14** exactly how they're in the [comment](https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1235/files#r34352085), these are set [meta types](https://duck.co/duckduckhack/metadata#category)

Line 14/15 
````perl
category "special";
topics "special_interest", "geek";
````

**Comment by GuiltyDolphin** [comment]
This is really old... I'll close this for now.

@abhijainn27 If you want to finish this, just let us know.

**Pull request closed without merging**